src/actions/repeated-action.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2011 - 2020 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This file is part of Touchégg.
+ *
+ * Touchégg is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License  as  published by  the  Free Software
+ * Foundation,  either version 3 of the License,  or (at your option)  any later
+ * version.
+ *
+ * Touchégg is distributed in the hope that it will be useful,  but  WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the  GNU General Public License  for more details.
+ *
+ * You should have received a copy of the  GNU General Public License along with
+ * Touchégg. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "actions/repeated-action.h"
+
+void RepeatedAction::executePrelude() {}
+void RepeatedAction::executePostlude() {}
+
+void RepeatedAction::onGestureBegin(const Gesture &gesture) {
+  // read repeated action settings
+  if (this->settings.count("repeat") == 1) {
+    this->repeat = (this->settings.at("repeat") == "true");
+  }
+
+  if (this->settings.count("on") == 1) {
+    this->onBegin = (this->settings.at("on") == "begin");
+  }
+
+  // execute supplied prelude
+  this->executePrelude();
+
+  // run action if required
+  if (!this->repeat && this->onBegin) {
+    if (gesture.percentage() > this->threshold) {
+      this->executeAction(gesture);
+      this->onBeginExecuted = true;
+    }
+  }
+}
+
+void RepeatedAction::onGestureUpdate(const Gesture &gesture) {
+  if (this->onBegin && !this->onBeginExecuted) {
```

Comment:
Why do you execute the `onBegin` action here again?

Reply:
See above, it's executed here if the threshold wasn't passed on gesture begin

Reply:
fix: 86170b44b507cc72077c369d560298a83c57fc2b